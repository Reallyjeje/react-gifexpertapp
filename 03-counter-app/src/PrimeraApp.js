// import React, { Fragment } from 'react';

import React from 'react';

// FUNCIONAL COMPONENTS

const PrimeraApp = ( props ) => {


const saludo = 'Hola, soy goku'

console.log(props);

    return (
        <>
           <h1> { saludo }!!!? </h1>
           {/* <pre> { JSON.stringify( saludo, null, 3 ) } </pre> */}
           <p> { subtitulo } </p>
        </>
    );
 
    


}



export default PrimeraApp;