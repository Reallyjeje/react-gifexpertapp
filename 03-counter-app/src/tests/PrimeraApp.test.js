import React from 'react';
import '@testing-library/jest-dom'
import PrimeraApp from '../PrimeraApp';
import { shallow } from 'enzyme'


describe('pruebas en <PrimeraApp />', () => {

//     test('debe de mostrar el mensaje "Hola, soy goku"', () => {
//         const saludo = 'Hola, soy goku';
//         const { getByText } = render( <PrimeraApp saludo={saludo} /> )
//         expect( getByText(saludo) ).toBeInTheDocument();
//     })


    test('debe de mostrar <PrimeraApp /> correctamente', () => {

        const saludo = 'Hola, soy goku';

        const wrapper = shallow( <PrimeraApp saludo={ saludo } /> )


        expect( wrapper ).toMatchSnapshot();

    })

    test('debe de mostrar el subtitulo enviado por props', () => {



        const saludo = 'Hola, soy goku';
        const subTitulo = 'soy un subtitulo';
        const  wrapper = shallow( 
        <PrimeraApp
            saludo={ saludo } 
            subtitulo={ subTitulo }
        /> )

        const textoParrafo = wrapper.find('p').texto();
        // console.log(textoParrafo);

        expect( textoParrafo ).toBe( subTitulo );



    })





})


