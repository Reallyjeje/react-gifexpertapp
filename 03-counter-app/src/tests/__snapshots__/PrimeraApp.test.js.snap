// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <PrimeraApp /> debe de mostrar <PrimeraApp /> correctamente 1`] = `
<Fragment>
  <h1>
     
    Hola, soy goku
    !!!? 
  </h1>
  <p>
    Mi primera aplicaci√≥n
  </p>
</Fragment>
`;
